apps/els_core/src/els_provider.erl

```diff
@@ -48,7 +48,9 @@
 -type request()  :: {atom() | binary(), map()}.
 -type state() :: #{ in_progress := [progress_entry()]
                   , in_progress_diagnostics := [diagnostic_entry()]
+                  , open_buffers := [buffer()]
```

Comment:
Sure, I can switch to a set. The number of open files is generally not huge but it makes sense, especially given the deletion logic.

Reply:
Yeah, my concern was more about the clarity of the code (seeing the extra `lists:delete` calls) than performance either. I agree that this probably won't have a significant perf impact since the list will be small anyway.